dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/view/UIDMapper.java

```diff
@@ -25,24 +25,15 @@
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
-package org.hisp.dhis.tracker.imports.validation.validator.event;
+package org.hisp.dhis.webapi.controller.tracker.view;
 
-import static org.hisp.dhis.tracker.imports.validation.validator.ValidationUtils.checkUidFormat;
-import static org.hisp.dhis.tracker.imports.validation.validator.ValidationUtils.validateNotesUid;
+import org.hisp.dhis.common.UID;
+import org.mapstruct.Mapper;
 
-import org.hisp.dhis.tracker.imports.bundle.TrackerBundle;
-import org.hisp.dhis.tracker.imports.domain.Event;
-import org.hisp.dhis.tracker.imports.validation.Reporter;
-import org.hisp.dhis.tracker.imports.validation.Validator;
+@Mapper
+public interface UIDMapper {
 
-/**
- * @author Morten Svanæs <msvanaes@dhis2.org>
- */
-class UidValidator implements Validator<Event> {
-  @Override
-  public void validate(Reporter reporter, TrackerBundle bundle, Event event) {
-    checkUidFormat(event.getEvent(), reporter, event, event, event.getEvent());
-
-    validateNotesUid(event.getNotes(), reporter, event);
+  default UID fromString(String stringUid) {
+    return stringUid == null ? null : UID.of(stringUid);
```

Comment:
I would assume we could directly map UIDs without any extra mapstruct mapper. Like UID being used by Spring without extra Spring code.

Do we need this Mapper only to accept `null`? When are `null`s allowed? Maybe document this in a javadoc.